src/App.js

```diff
@@ -1,12 +1,43 @@
 import React, { Component } from 'react';
+
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export class App extends Component {
-  state = {};
+  state = {
+    movies: moviesFromServer,
+    query: '',
+  };
+
+  changeHandler = (event) => {
+    this.setState({
+      query: event.target.value,
+    });
+  }
+
+  getVisibleMovies = ({ movies, query }) => {
+    let visibleMovies = [...movies];
+
+    if (query) {
+      const lowerQuery = query.toLowerCase();
+
+      visibleMovies = visibleMovies
+        .filter(movie => (
```

Comment:
```suggestion
    if (query) {
      const lowerQuery = query.toLowerCase();

      visibleMovies = movies.filter(movie => (
```
'filter' is a not-mutating method. Why do you create a copy?